Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -243,6 +243,26 @@ function Get-ServiceFabricApplicationUpgradeAction
     return Get-ServiceFabricApplicationUpgrade -ApplicationName $ApplicationName
 }
 
+function Wait-ServiceFabricApplicationUpgradeAction
+{
+    Param (
+        [string]
+        $ApplicationName
+    )
+
+    $global:operationId = $SF_Operations.WaitApplicationUpgradeStatus
+
+    Write-Host (Get-VstsLocString -Key SFSDK_WaitingForUpgrade)
+    $upgradeStatusFetcher = { Get-ServiceFabricApplicationUpgrade -ApplicationName $ApplicationName }
+    $upgradeRetryEvaluator = { param($upgradeStatus) return ($upgradeStatus.UpgradeState -ne "RollingBackCompleted" -and $upgradeStatus.UpgradeState -ne "RollingForwardCompleted") }
+    return Invoke-ActionWithRetries -Action $upgradeStatusFetcher `
+        -ResultRetryEvaluator $upgradeRetryEvaluator `
+        -MaxTries 2147483647 `
+        -RetryIntervalInSeconds 3 `
+        -RetryableExceptions @("System.Fabric.FabricTransientException") `
```

Comment:
Should we add TimeoutException here as well.

Reply:
Adding timeout exception is not a harm. Adding!